File: mobile/lib/domain/entities/safety_check_item.dart
/// 안전점검 항목 엔티티 (V1)
class SafetyCheckItem {
  /// 문항 내 순서
  final int seqNum;

  /// 점검 내용 텍스트
  final String contents;

  const SafetyCheckItem({
    required this.seqNum,
    required this.contents,
  });

  @override
  bool operator ==(Object other) {
    if (identical(this, other)) return true;
    return other is SafetyCheckItem &&
        other.seqNum == seqNum &&
        other.contents == contents;
  }

  @override
  int get hashCode => Object.hash(seqNum, contents);

  @override
  String toString() =>
      'SafetyCheckItem(seqNum: $seqNum, contents: $contents)';
}

File: backend/src/main/kotlin/com/otoki/internal/safetycheck/entity/SafetyCheckItem.kt
package com.otoki.internal.safetycheck.entity

import com.otoki.internal.common.entity.BaseEntity
import com.otoki.internal.common.salesforce.HCColumn
import com.otoki.internal.common.salesforce.HCTable
import jakarta.persistence.*

@Entity
@Table(name = "safetycheck_list")
@HCTable("safetycheck_list")
@IdClass(SafetyCheckItemId::class)
class SafetyCheckItem(

    @Id
    @Column(name = "question_num", nullable = false)
    @HCColumn("question_num")
    val questionNum: Int = 0,

    @Id
    @Column(name = "seq_num", nullable = false)
    @HCColumn("seq_num")
    val seqNum: Int = 0,

    @Column(name = "contents", nullable = false, length = 500)
    @HCColumn("contents")
    val contents: String = "",

    @Column(name = "use_yn", length = 1)
    @HCColumn("use_yn")
    val useYn: String? = "Y"

    // Phase2: 기존 V2 필드 주석 처리
    // val id: Long = 0,                    // auto-increment PK → 복합 키로 대체
    // val categoryId: Long = 0,            // V1에 없음
    // val label: String,                   // → contents로 대체
    // val sortOrder: Int,                  // V1에 없음
    // val required: Boolean = true,        // V1에 없음
    // val active: Boolean = true           // → useYn (String 'Y'/'N')으로 대체
) : BaseEntity()
